### PR TITLE
RavenDB-13411 Upgrade 3.5's owin to latest version

### PR DESCRIPTION
**Pull request by maximburyak**
It's a big change, but the OWIN update is really needed

### CODE REVIEW

File: Raven.Tests.FileSystem/Raven.Tests.FileSystem.csproj [hunk 1]
```diff
@@ -74,9 +74,9 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\packages\FluentAssertions.4.0.0\lib\net45\FluentAssertions.Core.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Owin.Host.HttpListener, Version=3.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\Microsoft.Owin.Host.HttpListener.3.0.1\lib\net45\Microsoft.Owin.Host.HttpListener.dll</HintPath>
+    <Reference Include="Microsoft.Owin.Host.HttpListener, Version=4.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Owin.Host.HttpListener.4.0.1\lib\net45\Microsoft.Owin.Host.HttpListener.dll</HintPath>
+      <Private>True</Private>
```
Comment (ppekrol) [inline Raven.Tests.FileSystem/Raven.Tests.FileSystem.csproj]: double 'Private'

File: Raven.Tests.Web/Areas/HelpPage/ApiDescriptionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+using System;
```
Comment (ppekrol) [inline Raven.Tests.Web/Areas/HelpPage/ApiDescriptionExtensions.cs]: those files need to be removed

File: Raven.Tests.Web/Raven.Tests.Web.csproj [hunk 3]
```diff
@@ -152,6 +144,34 @@
     <Compile Include="App_Start\FilterConfig.cs" />
     <Compile Include="App_Start\RouteConfig.cs" />
     <Compile Include="App_Start\WebApiConfig.cs" />
+    <Compile Include="Areas\HelpPage\ApiDescriptionExtensions.cs" />
```
Comment (ppekrol) [inline Raven.Tests.Web/Raven.Tests.Web.csproj]: Needs to be removed

File: Raven.Tests.FileSystem/Raven.Tests.FileSystem.csproj (hunk 1 above)
Comment (maximburyak) [reply Raven.Tests.FileSystem/Raven.Tests.FileSystem.csproj]: done

File: Raven.Tests.Web/Areas/HelpPage/ApiDescriptionExtensions.cs (hunk 2 above)
Comment (maximburyak) [reply Raven.Tests.Web/Areas/HelpPage/ApiDescriptionExtensions.cs]: done

File: Raven.Tests.Web/Raven.Tests.Web.csproj (hunk 3 above)
Comment (maximburyak) [reply Raven.Tests.Web/Raven.Tests.Web.csproj]: done

**Pull request merged**